Update definition and scope note for gist:Specification (#1405)
semanticarts/gist: docs/release_notes/1404-update-definition-of-gistspecification.md

### Minor Updates

> - Updated definition, example, scope note of `gist:Specification` to explicitly specify common ways of using the class and to disabmiguate with `gist:KnowledgeConcept`. Issue [#1404](https://github.com/semanticarts/gist/issues/1404).

Reviewer: ```suggestion
- Updated definition, example, scope note of `gist:Specification` to explicitly specify common ways of using the class and to distinguish from `gist:KnowledgeConcept`. Issue [#1404](https://github.com/semanticarts/gist/issues/1404).
```
